src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -816,6 +816,8 @@ private async Task EnsureInitializeAsync()
                         return;
                     }
 
+                    TelemetryActivity.NuGetTelemetryService = new NuGetVSTelemetryService();
```

Comment:
This isn't going to initialize the telemetry service sooner in all scenarios, it'll be much later in certain scenarios.  Maybe the alternative is to make sure that the assignment happens in all `AsyncPackage` implementations. There's only one I know of that isn't in this PR's diff: `src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs`. But you should search for other classes that extend `AsyncPackage` to see if there are any others.

Reply:
In my testing, package initialization was happening after restore started in the background following solution load.  I set a breakpoint and this was the earliest that I saw telemetry being logged.  Can you give me an idea of scenarios I can try out?

Reply:
Try a solution that only has "legacy" csproj (no SDK style projects). Or, a solution with only C++ projects, and open PM UI without building first). Also, open Package Manger Console, close VS, then launch again. NuGetPackage, and PMC, should be created before VsSolutionManager's EnsureInitiailizeAsync.  I'm just not sure which of these scenarios will attempt to send telemetry before VsSolutionManager runs EnsureInitializeAsync. Both PM UI and PMC need the project list, which needs VsSolutionManager.  Our Tools->Options page doesn't need VsSolutionManager, but I don't think we have any telemetry there.

A more obscure scenario is to open VS, open an empty solution (a solution without SDK style projects probably works as well, I just know that an SDK style project is trigger nomination which triggers restore, which therefore initializes VsSolutionManager), right click the solution, so that the context menu is loaded, then quit VS.  Erick added PowerShell host telemetry, but this scenario probably did not trigger VsSolutionManager, and therefore telemetry to tell us that PowerShell was not used will not be sent.

`AsyncPackage.InitializeAsync` is the `Program.Main` of VS extensions, but we have multiple classes implementing `AsyncPackage` for different features (to reduce perf impact & memory usage). So, if we want to make sure that this is initialized as early as possible, then we need to find all of these entry points.

Reply:
What if we initialized the service in the package constructor? 


Would that be early enough?

Reply:
If it happened in the RestoreManagerPackage constructor, I think that would work because I **think** that package is always loaded, no matter what.  But there are definately scenarios where NuGetPackage (in NuGet.Tools.csproj) is never loaded, so loading it in that class constructor won't work.

Reply:
Another scenario to test is load a solution without any projects that support NuGet. For example, load a solution with python or node.js projects. Open then Project menu. I think when VS tries to figure out whether or not to enable the "Manage NuGet Packages" menu item it will load NuGetPackage, but VsSolutionManager's EnsureInitializedAsync method will probably never be called.

Reply:
These are great suggestions, I've already found that my current solution didn't do the trick.  I'm expanding it and having better luck.  I'll force push what I have.  This is what works so far:
- [x] Launch Package Manager Console before opening solution
  Telemetry is initialized by `NuGetPackage.InitializeAsync()`
- [x] Launch NuGet Manager Settings before opening solution
  Telemetry is initialized by `NuGetPackage.InitializeAsync()`
- [x] Load solution with legacy CSPROJ and wait for background restore
  Telemetry is initialized by `NuGetPackage.InitializeAsync()`
- [x] Load solution with CPS 
  Telemetry is initialized by `RestoreManagerPackage.InitializeAsync()`
- [x] Load blank solution
  Telemetry is initialized by `RestoreManagerPackage.InitializeAsync()`
- [x] Create solution with Python project via start page / new project
  Telemetry is initialized by `RestoreManagerPackage.InitializeAsync()`

So far it looks like all that was missing was initializing the telemetry service in `RestoreManagerPackage`